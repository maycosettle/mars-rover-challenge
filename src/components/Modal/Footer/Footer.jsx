import PropTypes from 'prop-types';

import useStyles from './styles';

const Footer = ({ onClose, onSubmit }) => {
  const styles = useStyles();

  return (
    <div className={styles.container}>
      <button type="button" className={styles.confirm} onClick={onSubmit}>
        Confirm
      </button>
      <button type="button" className={styles.cancel} onClick={onClose}>
        Cancel
      </button>
    </div>
  );
};

Footer.propTypes = {
  onClose: PropTypes.func,
  onSubmit: PropTypes.func,
};

Footer.defaultProps = {
  onClose: () => {},
  onSubmit: () => {},
};

export default Footer;
