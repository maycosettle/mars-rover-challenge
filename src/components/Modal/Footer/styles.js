const useStyles = () => ({
  container: 'mt-5 sm:mt-6 sm:grid sm:grid-cols-2 sm:gap-3 sm:grid-flow-row-dense',
  confirm:
    'w-full inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-primary-600 text-base font-medium text-white hover:bg-primary-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500 sm:col-start-2 sm:text-sm',
  cancel:
    'mt-3 w-full inline-flex justify-center rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-gray-700 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500 sm:mt-0 sm:col-start-1 sm:text-sm',
});

export default useStyles;
