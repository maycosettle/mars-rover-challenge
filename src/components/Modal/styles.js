const useStyles = () => ({
  mainContainer: 'fixed z-10 inset-0 overflow-y-auto',
  container: 'flex items-end justify-center min-h-screen pt-4 px-4 pb-20 text-center sm:block sm:p-0',
  overlay: 'fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity',
  children:
    'inline-block align-bottom bg-white rounded-lg px-4 pt-5 pb-4 text-left overflow-hidden shadow-xl transform transition-all sm:my-8 sm:align-middle sm:max-w-lg sm:w-full sm:p-6',
});

export default useStyles;
