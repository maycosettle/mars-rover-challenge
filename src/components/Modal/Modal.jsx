import PropTypes from 'prop-types';
import { Fragment } from 'react';
import { Dialog, Transition } from '@headlessui/react';

import { Footer } from './Footer';

import useStyles from './styles';

const Modal = ({ openModal, onClose, onSubmit, children }) => {
  const styles = useStyles();

  return (
    <Transition.Root show={openModal} as={Fragment}>
      <Dialog as="div" className={styles.mainContainer} onClose={onClose}>
        <div className={styles.container}>
          <Transition.Child
            as={Fragment}
            enter="ease-out duration-300"
            enterFrom="opacity-0"
            enterTo="opacity-100"
            leave="ease-in duration-200"
            leaveFrom="opacity-100"
            leaveTo="opacity-0"
          >
            <Dialog.Overlay className={styles.overlay} />
          </Transition.Child>

          <Transition.Child
            as={Fragment}
            enter="ease-out duration-300"
            enterFrom="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
            enterTo="opacity-100 translate-y-0 sm:scale-100"
            leave="ease-in duration-200"
            leaveFrom="opacity-100 translate-y-0 sm:scale-100"
            leaveTo="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
          >
            {children && (
              <div className={styles.children}>
                {children}
                <Footer onClose={onClose} onSubmit={onSubmit} />
              </div>
            )}
          </Transition.Child>
        </div>
      </Dialog>
    </Transition.Root>
  );
};

Modal.propTypes = {
  openModal: PropTypes.func,
  onClose: PropTypes.func,
  onSubmit: PropTypes.func,
  children: PropTypes.node,
};

Modal.propTypes = {
  openModal: () => {},
  onClose: () => {},
  onSubmit: () => {},
};

export default Modal;
