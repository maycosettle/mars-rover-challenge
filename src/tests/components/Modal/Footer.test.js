import { shallow } from 'enzyme';
import { Footer } from 'components/Modal/Footer';

describe('tests on <Footer />', () => {
  const wrapper = shallow(<Footer />);

  test('Should render Footer correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
