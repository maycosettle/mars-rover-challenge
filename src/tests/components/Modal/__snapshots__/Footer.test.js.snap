// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <Footer /> Should render Footer correctly 1`] = `
<div
  className="mt-5 sm:mt-6 sm:grid sm:grid-cols-2 sm:gap-3 sm:grid-flow-row-dense"
>
  <button
    className="w-full inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-primary-600 text-base font-medium text-white hover:bg-primary-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500 sm:col-start-2 sm:text-sm"
    onClick={[Function]}
    type="button"
  >
    Confirm
  </button>
  <button
    className="mt-3 w-full inline-flex justify-center rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-gray-700 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500 sm:mt-0 sm:col-start-1 sm:text-sm"
    onClick={[Function]}
    type="button"
  >
    Cancel
  </button>
</div>
`;
